### PR TITLE
Add userId to questionOptions

### PR DESCRIPTION
**Pull request by MartinBenediktBusch**
This PR adds the `userId` to the `questionOptions` table. Note that the reference to the `userId` in the question options table can be `null` as this allows us to keep inserting custom options into the `questionOptions` table. Furthermore, note that the userId gets retrieved and uptaded in the `questionOptions` table upon triggering `saveRegistration`.

### CODE REVIEW

File: src/db/questionOptions.ts [hunk 1]
```diff
@@ -20,6 +22,10 @@ export const questionOptions = pgTable('question_options', {
 });
 
 export const questionOptionsRelations = relations(questionOptions, ({ one, many }) => ({
+  user: one(users, {
+    fields: [questionOptions.userId],
+    references: [users.id],
+  }),
```
Comment (diegoalzate) [inline src/db/questionOptions.ts:28]: add relation on user as well

**Review by diegoalzate — CHANGES_REQUESTED** [review]
check comments

File: src/db/questionOptions.ts (hunk 1 above)
Comment (MartinBenediktBusch) [reply src/db/questionOptions.ts:28]: added relation to `users` table as well.

File: src/db/users.ts [hunk 2]
```diff
@@ -26,6 +27,7 @@ export const usersRelations = relations(users, ({ many, one }) => ({
   federatedCredential: one(federatedCredentials),
   comments: many(comments),
   likes: many(likes),
+  questionOptions: one(questionOptions),
```
Comment (diegoalzate) [inline src/db/users.ts:30]: this will have many question options

**Review by diegoalzate — APPROVED** [review]

File: src/db/users.ts (hunk 2 above)
Comment (MartinBenediktBusch) [reply src/db/users.ts]: done.

**Pull request merged**